### DISCUSSION TITLE
Product Info page behavior when DISABLED_PRODUCTS_TRIGGER_HTTP200 = true

### DISCUSSION THREAD

**Discussion by scottcwilson** [zencart/zencart#6165] [Ideas]
If you set DISABLED_PRODUCTS_TRIGGER_HTTP200 to true (in Admin > Configuration > Stock), then view the product info page of a disabled product, it shows an add to cart button.  This may not be what you want.   (In fact, it's certainly what you want, because pressing Add to Cart doesn't actually do anything - the add to cart logic rejects the add of a disabled product).

I'm not sure whether this should be modified in the product_info template (or perhaps elsewhere) or simply documented in https://docs.zen-cart.com/user/products/products_status/ 

I only have one customer who uses this (relatively) new config so I just discovered this.

** Updated for clarification: the product is disabled but has products_quantity > 0.

**Comment by scottcwilson** [comment]
Easiest fix which will work for most stores that only sell product_type = 1 items:

change `includes/templates/YOUR_TEMPLATE/templates/tpl_product_info_display` from 
```
<!--bof Add to Cart Box -->
<?php
if (CUSTOMERS_APPROVAL == 3 and TEXT_LOGIN_FOR_PRICE_BUTTON_REPLACE_SHOWROOM == '') {
  // do nothing
```
to
```
<!--bof Add to Cart Box -->
<?php
if ((CUSTOMERS_APPROVAL == 3 and TEXT_LOGIN_FOR_PRICE_BUTTON_REPLACE_SHOWROOM == '') || ($product_info->fields['products_status'] != 1)) {
  // do nothing
```

**Comment by proseLA** [comment]
i am not able to reproduce this situation and therefore i am of the opinion that this is just not true.  

i think there is another config button which is allowing the add to cart button to appear.  it has nothing to do with the `DISABLED_PRODUCTS_TRIGGER_HTTP200`.

this constant correctly issues the http status it states in the description.  and if set to true, the product info will appear.  if set to false, a product not found page appears.

using the test data i have a sold out button where the add to cart button should be indicating that a different config value is controlling that behavior.

**Comment by scottcwilson** [comment]
Please re-test with an item that has stock (products_quantity > 0).

**Comment by proseLA** [comment]
the add to cart button does come up.

why someone would have a disabled product with a quantity > 0 is perhaps a question you can answer.

**Comment by scottcwilson** [comment]
All kinds of reasons.  Safety recall.   Legislative change.  Problems with supplier.   
Some people don't track stock at all but set a default quantity when they create products.  All kinds of configurations are possible. 
Thanks for taking the time to test.

**Comment by scottcwilson** [comment]
Not that much enthusiasm for this change - moving to discussion.
